File: archive/sdk/python/examples/basic_usage.py
"""Basic usage example for PAW Python SDK."""

import asyncio
from paw import PawClient, PawWallet, ChainConfig


async def main():
    """Main example function."""
    # 1. Create a new wallet
    mnemonic = PawWallet.generate_mnemonic()
    print(f"Generated mnemonic: {mnemonic}")

    wallet = PawWallet("paw")
    wallet.from_mnemonic(mnemonic)
    print(f"Wallet address: {wallet.address}")

    # 2. Connect to the blockchain
    config = ChainConfig(
        rpc_endpoint="http://localhost:26657",
        rest_endpoint="http://localhost:1317",
        chain_id="paw-mvp-1"
    )

    async with PawClient(config) as client:
        await client.connect_wallet(wallet)
        print(f"Connected to chain: {await client.get_chain_id()}")

        # 3. Check balance
        balance = await client.bank.get_balance(wallet.address, "upaw")
        if balance:
            print(f"Balance: {client.bank.format_balance(balance)}")

        # 4. Get all balances
        all_balances = await client.bank.get_all_balances(wallet.address)
        print(f"All balances: {len(all_balances)} tokens")


if __name__ == "__main__":
    asyncio.run(main())

File: archive/sdk/javascript/examples/basic-usage.ts
import { PawClient, PawWallet } from '@paw-chain/sdk';

async function main() {
  // 1. Create a new wallet
  const mnemonic = PawWallet.generateMnemonic();
  console.log('Generated mnemonic:', mnemonic);

  const wallet = new PawWallet('paw');
  await wallet.fromMnemonic(mnemonic);

  const address = await wallet.getAddress();
  console.log('Wallet address:', address);

  // 2. Connect to the blockchain
  const client = new PawClient({
    rpcEndpoint: 'http://localhost:26657',
    restEndpoint: 'http://localhost:1317',
    chainId: 'paw-mvp-1'
  });

  await client.connectWithWallet(wallet);
  console.log('Connected to chain:', await client.getChainId());

  // 3. Check balance
  const balance = await client.bank.getBalance(address, 'upaw');
  console.log('Balance:', balance);

  // 4. Get all balances
  const allBalances = await client.bank.getAllBalances(address);
  console.log('All balances:', allBalances);

  // 5. Disconnect
  await client.disconnect();
  console.log('Disconnected');
}

main().catch(console.error);
